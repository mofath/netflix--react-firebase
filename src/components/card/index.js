import React, { useState, useContext, createContext } from 'react';

import { Container, Group, Title, SubTitle, Text, Item, Image, Enteties, Maturity, Content, FeatureClose, FeatureText, Feature, Meta, FeatureTitle } from './styles/card'

export const FeatureContext = createContext();

export default function Card({ children, ...restProps }) {
    const [ShowFeature, setShowFeature] = useState(false);
    const [ItemFeature, setItemFeature] = useState({});

    return (
        <FeatureContext.Provider value={{ ShowFeature, setShowFeature, ItemFeature, setItemFeature }}>
            <Container {...restProps}>{children}</Container>
        </FeatureContext.Provider>
    )
};

Card.Group = function cardGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>
};

Card.Title = function cardTitle({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>
};

Card.SubTitle = function cardSubTitle({ children, ...restProps }) {
    return <SubTitle {...restProps}>{children}</SubTitle>
};

Card.Text = function cardText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>
};

Card.Meta = function cardMeta({ children, ...restProps }) {
    return <Meta {...restProps}>{children}</Meta>
};


Card.Item = function CardItem({ item, children, ...restProps }) {
    const { setShowFeature, setItemFeature } = useContext(FeatureContext);

    return (
        <Item
            onClick={() => {
                setItemFeature(item);
                setShowFeature(true);
            }}
            {...restProps}
        >
            {children}
        </Item>
    );
};

Card.Enteties = function cardEnteties({ children, ...restProps }) {
    return <Enteties {...restProps} >{children}</Enteties>
}

Card.Image = function cardImage({ src, ...restProps }) {
    return <Image src={src} {...restProps} />
}

Card.Maturity = function cardMaturity({ children, ...restProps }) {
    return <Maturity {...restProps} >{children}</Maturity>
}

Card.Content = function cardContent({ children, ...restProps }) {
    return <Content {...restProps} >{children}</Content>
}

Card.Feature = function CardFeature({ children, category, ...restProps }) {
    const { ShowFeature, setShowFeature, ItemFeature } = useContext(FeatureContext);

    return ShowFeature ? (
        <Feature {...restProps} src={`images/${category}/${ItemFeature.genre}/${ItemFeature.slug}/large.jpg`} >
            <Content>
                <FeatureTitle>{ItemFeature.title}</FeatureTitle>
                <FeatureText>{ItemFeature.description}</FeatureText>
                <FeatureClose onClick={() => setShowFeature(false)}>
                    <img src='/images/icons/close.png' alt="Close" />
                </FeatureClose>
                <Group margin="30px 0" flexDirection="row" alignItems="center">
                    <Maturity rating={ItemFeature.maturity}>
                        {ItemFeature.maturity < 12 ? 'PG' : ItemFeature.maturity}
                    </Maturity>
                    <FeatureText fontWeight="bold">
                        {ItemFeature.genre.charAt(0).toUpperCase() + ItemFeature.genre.slice(1)}
                    </FeatureText>
                </Group>
                {children}
            </Content>
        </Feature >
    ) :
        null
}