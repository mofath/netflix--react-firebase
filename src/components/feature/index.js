import React from 'react';

import { Container, Title, SubTitle } from './styles/feature';


export default function Feature({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
}

Feature.Title = function featureTitle({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>
}

Feature.SubTitle = function featureSubTitle({ children, ...restProps }) {
    return <SubTitle {...restProps}>{children}</SubTitle>
}