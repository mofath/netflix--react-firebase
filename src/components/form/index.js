import React from 'react';

import { Container, Title, Text, SubText, Error, Link, Input, Submit, Base } from './styles/from';


export default function Form({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
};

Form.Input = function fromInput({ children, ...restProps }) {
    return <Input {...restProps}>{children}</Input>;
};

Form.Submit = function fromSubmit({ children, ...restProps }) {
    return <Submit {...restProps}>{children}</Submit>;
};

Form.Error = function fromError({ children, ...restProps }) {
    return <Error {...restProps}>{children}</Error>;
};

Form.Base = function fromBase({ children, ...restProps }) {
    return <Base {...restProps}>{children}</Base>;
};


Form.Title = function fromTitle({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>;
};

Form.Text = function fromText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>;
};

Form.SubText = function fromSubText({ children, ...restProps }) {
    return <SubText {...restProps}>{children}</SubText>;
};

Form.Link = function fromLink({ children, ...restProps }) {
    return <Link {...restProps}>{children}</Link>;
};