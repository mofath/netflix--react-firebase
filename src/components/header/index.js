import React, { useState } from 'react';

import { Link as ReachRouterLink } from 'react-router-dom';

import { Background, Container, Logo, ButtonLink, Feature, Text, FeatureCallOut, Link, Group, Picture, Dropdown, Profile, Search, SearchIcon, SearchInput } from './styles/header'

export default function Header({ bg = true, children, ...restProps }) {
    return bg ? (
        <Background data-testid="header-bg" {...restProps}>
            {children}
        </Background>
    ) : (
            <Container>
                {children}
            </Container>
        );
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
    return <Feature>{children}</Feature>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>;
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
    return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
    return (
        <ReachRouterLink to={to}>
            <Logo {...restProps} />
        </ReachRouterLink>
    );
};


Header.Text = function HeaderText({ children, ...restProps }) {
    return <Text {...restProps}>{children}</Text>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({ children, ...restProps }) {
    return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.TextLink = function headerTextLink({ children, ...restProps }) {
    return <Link {...restProps}>{children}</Link>
}

Header.Group = function headerGroup({ children, ...restProps }) {
    return <Group {...restProps}>{children}</Group>
}

Header.Picture = function headerPicture({ src, ...restProps }) {
    return <Picture {...restProps} src={`/images/users/${src}.png`} />
}

Header.Dropdown = function headerDropdown({ children, ...restProps }) {
    return <Dropdown {...restProps}>{children}</Dropdown>
}

Header.Profile = function headerProfile({ children, ...restProps }) {
    return <Profile {...restProps}>{children}</Profile>
}

Header.Search = function HeaderSearch({ searchTerm, setSearchTerm, ...restProps }) {
    const [SearchActive, setSearchActive] = useState(false);

    return (
        <Search {...restProps} >
            <SearchIcon onClick={() => setSearchActive(SearchActive => !SearchActive)} >
                <img src='images/icons/search.png' alt="search" />
            </SearchIcon>
            <SearchInput
                value={searchTerm}
                onChange={({ target }) => setSearchTerm(target.value)}
                placeholder="Search films and series"
                active={SearchActive}
            />
        </Search>
    )
}