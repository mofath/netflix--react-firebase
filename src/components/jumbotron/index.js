import React from 'react';

import { Inner, Item, Container, Pane, Title, Subtitle, Image } from './styles/jumbotron';

export default function Jumbotron({ children, direction = 'row', ...restProps }) {
    return (
        <Item {...restProps}>
            <Inner direction={direction} >
                {children}
            </Inner>
        </Item>
    )
}

Jumbotron.Container = function JumbotronConainer({ children, ...restProps }) {
    return <Container {...restProps}>{children}</Container>
}

Jumbotron.Pane = function JumbotronConainer({ children, ...restProps }) {
    return <Pane {...restProps}>{children}</Pane>
}


Jumbotron.Title = function JumbotronConainer({ children, ...restProps }) {
    return <Title {...restProps}>{children}</Title>
}

Jumbotron.Subtitle = function JumbotronConainer({ children, ...restProps }) {
    return <Subtitle {...restProps}>{children}</Subtitle>
}

Jumbotron.Image = function JumbotronConainer({ ...restProps }) {
    return <Image  {...restProps} />
}

