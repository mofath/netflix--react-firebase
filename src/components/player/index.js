import React, { useState, createContext, useContext } from 'react';

import { Cotainer, Inner, Button, Overlay, Close } from './styles/player';
import ReactDOM from 'react-dom';

export const PlayerContext = createContext();

export default function Player({ children, ...restProps }) {
    const [ShowPlayer, setShowPlayer] = useState(false);
    return (
        <PlayerContext.Provider value={{ ShowPlayer, setShowPlayer }}>
            <Cotainer {...restProps}>{children}</Cotainer>
        </PlayerContext.Provider>
    )
}

Player.Video = function PlayerVideo({ src, ...restProps }) {
    const { ShowPlayer, setShowPlayer } = useContext(PlayerContext)
    return ShowPlayer ? ReactDOM.createPortal(
        <Overlay onClick={() => setShowPlayer(false)} data-testid="player">
            <Inner>
                <video id="netflix-player" controls>
                    <source src={src} type="video/mp4" />
                </video>
                <Close />
            </Inner>
        </Overlay>,
        document.body
    )
        : null;
};

Player.Button = function PlayerButton({ ...restProps }) {
    const { setShowPlayer } = useContext(PlayerContext);

    return (
        <Button onClick={() => setShowPlayer((ShowPlayer) => !ShowPlayer)} {...restProps}>
            Play
        </Button>
    );
};