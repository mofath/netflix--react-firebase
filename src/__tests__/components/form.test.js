import React from "react";
import { render } from "@testing-library/react";
import { Form } from "../../components";

jest.mock("react-router-dom");

describe("<Form />", () => {
  let container, getByText, queryByText, getByPlaceholderText;
  let error = null;

  it("renders the <Form /> component with populated data", () => {
    const { container, getByText, getByPlaceholderText } = render(
      <Form>
        <Form.Title>Sign In Now</Form.Title>
        {error && <Form.Error>Email is already taken</Form.Error>}
        <Form.Base>
          <Form.Input placeholder="Email address" onChange={() => {}} />
          <Form.Input
            type="password"
            autoComplete="off"
            placeholder="Password"
            onChange={() => {}}
          />
          <Form.Submit disabled type="submit">
            Sign In
          </Form.Submit>
        </Form.Base>
        <Form.Text>
          New to Netflix? <Form.Link to="/signup">Sign up now.</Form.Link>
        </Form.Text>
        <Form.SubText>This page is protected by Google.</Form.SubText>
      </Form>
    );
    expect(getByText("This page is protected by Google.")).toBeTruthy();
    expect(getByText("Sign In Now")).toBeTruthy();
    expect(getByText("Sign In")).toBeTruthy();
    expect(getByText("Sign In").disabled).toBeTruthy();
    expect(getByPlaceholderText("Email address")).toBeTruthy();
    expect(getByPlaceholderText("Password")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <Form /> with an error", () => {
    const { container, getByText, queryByText } = render(
      <Form>
        <Form.Error>Email is already taken</Form.Error>
        <Form.Submit type="submit">Sign In</Form.Submit>
      </Form>
    );

    expect(getByText("Email is already taken")).toBeTruthy();
    expect(queryByText("Sign In").disabled).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
