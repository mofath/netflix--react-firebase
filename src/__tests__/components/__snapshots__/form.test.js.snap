// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders the <Form /> component with populated data 1`] = `
<div
  class="from__Container-sc-1x5h4v7-0 gDEElt"
>
  <h1
    class="from__Title-sc-1x5h4v7-5 iWjJRN"
  >
    Sign In Now
  </h1>
  <form
    class="from__Base-sc-1x5h4v7-1 gsXBYf"
  >
    <input
      class="from__Input-sc-1x5h4v7-2 Stbig"
      placeholder="Email address"
    />
    <input
      autocomplete="off"
      class="from__Input-sc-1x5h4v7-2 Stbig"
      placeholder="Password"
      type="password"
    />
    <button
      class="from__Submit-sc-1x5h4v7-3 cVEbht"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="from__Text-sc-1x5h4v7-6 kXgQwo"
  >
    New to Netflix? 
  </p>
  <p
    class="from__SubText-sc-1x5h4v7-7 htLfeT"
  >
    This page is protected by Google.
  </p>
</div>
`;

exports[`<Form /> renders the <Form /> with an error 1`] = `
<div
  class="from__Container-sc-1x5h4v7-0 gDEElt"
>
  <div
    class="from__Error-sc-1x5h4v7-4 bGebFr"
  >
    Email is already taken
  </div>
  <button
    class="from__Submit-sc-1x5h4v7-3 cVEbht"
    type="submit"
  >
    Sign In
  </button>
</div>
`;
